openc3/lib/openc3/models/tool_model.rb

```diff
@@ -155,15 +155,27 @@ def initialize(
     end
 
     def create(update: false, force: false)
+      tools = self.class.all(scope: @scope)
+
+      # Make sure a tool with this folder_name doesn't already exist
+      unless update
+        tools.each do |_tool_name, tool|
+          if tool['folder_name'] == @folder_name
+            raise "Tool with folder_name #{@folder_name} already exists at create"
```

Comment:
Who catches this and how does it present itself to the user?

Reply:
Its caught like all other errors that happen during plugin install.  Then presented to the user in installation status on the plugins tab.